uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -451,28 +451,41 @@ public static function findOverlappingLeaveRequests($contactID, $fromDate, $toDa
    *   And array of values from the Leave Request Status OptionGroup
    * @param bool $publicHolidaysOnly
    *   When true, will get the balance change only for the Public Holiday Leave Requests
+   * @param bool $expiredOnly
+   *   When true, will only count the balance change for expired days
    *
    * @return array
    */
   public static function getBalanceChangeByAbsenceType(
     $contactID,
     $periodID,
     $leaveRequestStatus = [],
-    $publicHolidaysOnly = false
+    $publicHolidaysOnly = false,
+    $expiredOnly = false
```

Comment:
Yeah, that would be the perfect scenario for us, but not for the frontend guys :D If we opt for the new API in leave_period_entitlement then they would need to implement a new service (a lot of code and new tests) to call this new API, while with the param option all they have to do is add it to the current service and add a couple of tests to it. There's also the fact that with a new API, they would need 2 calls to fetch the data. 

Another reason to have this as a single API call is the way the information will be displayed:
![selecao_002](https://cloud.githubusercontent.com/assets/388373/21768341/4ec1503a-d660-11e6-97d0-38c981e968a3.png)

As you can see, it's a single row displaying everything that has expired (currently Brought Forward + TOIL, but it can also include other things in the future).

So, what I'm trying to say is that even though I agree with you about separating things, from an API/frontend point of view it's better to keep them together, and the place that makes more sense to do that is on the LeaveRequest API (though it might be a bit confusing, most of the information comes from Leave Requests balance changes)